core-dictionary/src/main/java/org/apache/kylin/dict/global/AppendTrieDictionaryBuilder.java

```diff
@@ -86,7 +89,13 @@ public void addValue(String value) throws IOException {
         }
         checkState(sliceFileMap.firstKey().equals(AppendDictSliceKey.START_KEY), "first key should be \"\", but got \"%s\"", sliceFileMap.firstKey());
 
-        AppendDictSliceKey nextKey = sliceFileMap.floorKey(AppendDictSliceKey.wrap(valueBytes));
+        AppendDictSliceKey key = AppendDictSliceKey.wrap(valueBytes);
+        if (preKey != null && key.compareTo(preKey) < 0) {
```

Comment:
I'm afraid this check for each key will downgrade the performance of global dict building.

Reply:
This code is just used for verifying the issue. I can remove it later or add a flag for it.